### ISSUE TITLE
Touching outside of the cropping window moves the image on mobile

### ISSUE THREAD

**Issue by cpwood** [ValentinH/svelte-easy-crop#23]
I have a relative `crop-container` div surrounding the svelte-easy-crop component.

```svelte
        <div id="crop-container" bind:this={container} class="m-3" bind:clientWidth={width} bind:clientHeight={height}>
            <Cropper
                image={context.image}
                crop={{ x: 0, y: 0 }}
                zoom={2}
                aspect={17/6}
                maxZoom={8}
                showGrid={false}
                cropSize={{
                    height: height - 20,
                    width: width - 20
                }}
                on:cropcomplete={e => onCrop(e.detail.pixels)}
                />
        </div>
```
I've been testing zooming and panning on both desktop (macOS) and mobile (iPhone). 

  **Observed behaviour:** 

On desktop, if I click within the crop box and drag, it will pan the image. If I click outside the crop box (e.g. anywhere else on the web page), it doesn't interact with the image or cropper. This is also what I'd expect to happen.

However, on mobile, holding and dragging with one finger pans the image like with desktop, however trying to scroll the web page by pressing and dragging _outside_ of the crop box results in the image panning. This is inconsistent with desktop and not what I'd expect to happen. More frustratingly, I can't scroll up or down the page anymore.

  **Expected behaviour:** 

Interacting with the web page outside of the crop box doesn't impact the image or cropping area.

Is this behaviour by design or is it a bug? If it's by design, is there a way in which I can accomplish what I' trying to do?

Thanks in advance!

**Comment by cpwood** [comment]
I've spent a bit of time on this today and I think I've discovered the problem and have produced a fix (PR to follow).

## Mouse events

The general approach seems to be:

1. start monitoring for drag actions when a `mousedown` event occurs on the container`<div>` by setting `isDragging = true`;
2. receive `window`-level `mousemove` and `mouseup` events to track actual movement thereafter.
3. set `isDragging = false` when a `mouseup`event occurs and hence stops paying attention to `window`-level `mousemove` and `mouseup` events.

## Touch events

I believe the same general approach _should_ be followed, but it actually differs:

1. a `touchstart` event on the container `<div>` has **no impact** on `isDragging`;
2. receives `window`-level `touchmove` and `touchend` events to track touch movement **for the entire `window`** and sets `isDragging = true`.
3. set `isDragging = false` when a `touchend`event occurs and stops paying attention to `window`-level `touchend` events but **any later touch anywhere in the window** will start them listening again.

## Fix approach

In my own fork I have:

1. moved all `isDragging = true` instances to the first line of `onDragStart()`. This means `isDragging` is now set to `true` whether it's a mouse or touch event on the container `<div>`.
2. changed the handler for `onTouchMove()` so that instead of saying `isDragging = true` it checks whether it's _already_ true and returns straight away if it's `false` (mirroring the mouse movement code).

The tests all pass.

I'll submit a PR in a few moments!

**Comment by ValentinH** [comment]
Thanks for the detailed and well explained issue!

This broken behaviour was introduced in #17 when it was decided to count on Svelte to manage the global events. This was not a good idea for the mousemove/mouseup / touchmove/touchend events that were changed [here](https://github.com/ValentinH/svelte-easy-crop/pull/17/files?authenticity_token=Tj2jXKBMDCj2pW%2BALLEQnLHTVnmoG8Enzc%2Fue6e%2F6DRwMGyfWGmJHynaGZSZOup9KGhDVxuBzdZBM4lHTR47iQ%3D%3D&file-filters%5B%5D=.js&file-filters%5B%5D=.lock&file-filters%5B%5D=.md&file-filters%5B%5D=.svelte&file-filters%5B%5D=dotfile&w=1#diff-cfc4b5c0c7495f9b6fe276b630abd27ffd1fd7da7fe0f9d95780aa24df7c8161L97)

I think we should rollback this part and only attach these events once the user has actually started a gesture inside the cropper. This is how we do it on `react-easy-crop`.

**Comment by cpwood** [comment]
No problem! OK, will leave it with you.

**Comment by ValentinH** [comment]
This was fixed in v1.0.6
